src/Microsoft.NetCore.Analyzers/Core/Security/DoNotInstallRootCert.cs

```diff
@@ -0,0 +1,210 @@
+                                {
+                                    var argumentOperation = (IArgumentOperation)operationAnalysisContext.Operation;
+
+                                    if (x509TypeSymbol.Equals(argumentOperation.Parameter.Type))
+                                    {
+                                        lock (rootOperationsNeedingAnalysis)
+                                        {
+                                            rootOperationsNeedingAnalysis.Add((argumentOperation.GetRoot(), operationAnalysisContext.ContainingSymbol));
+                                        }
+                                    }
+                                },
+                                OperationKind.Argument);
+                        });
+
+                    compilationStartAnalysisContext.RegisterCompilationEndAction(
+                        (CompilationAnalysisContext compilationAnalysisContext) =>
+                        {
+                            PooledDictionary<(Location Location, IMethodSymbol Method), HazardousUsageEvaluationResult> allResults = null;
+
+                            try
+                            {
+                                lock (rootOperationsNeedingAnalysis)
+                                {
+                                    if (!rootOperationsNeedingAnalysis.Any())
+                                    {
+                                        return;
+                                    }
+
+                                    allResults = PropertySetAnalysis.BatchGetOrComputeHazardousUsages(
+                                        compilationAnalysisContext.Compilation,
+                                        rootOperationsNeedingAnalysis,
+                                        WellKnownTypeNames.SystemSecurityCryptographyX509CertificatesX509Store,
+                                        constructorMapper,
+                                        PropertyMappers,
+                                        hazardousUsageEvaluators,
+                                        InterproceduralAnalysisConfiguration.Create(
+                                            compilationAnalysisContext.Options,
+                                            SupportedDiagnostics,
+                                            defaultInterproceduralAnalysisKind: InterproceduralAnalysisKind.ContextSensitive,
+                                            cancellationToken: compilationAnalysisContext.CancellationToken));
+                                }
+
+                                if (allResults == null)
+                                {
+                                    return;
+                                }
+
+                                foreach (var kvp in allResults)
+                                {
+                                    if (kvp.Value.Equals(HazardousUsageEvaluationResult.Flagged))
+                                    {
```

Comment:
Do you want to do anything in the MaybeFlagged case?  I'm fine either way; your call.  Just want to make sure you considered it.